### PR TITLE
Pal 710 ecr push

### PR DESCRIPTION
**Pull request by m78233**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -120,45 +120,62 @@ timestamps {
...
             stage('Prerequisites') {
                 dir ('Palisade-common') {
                     git branch: 'develop', url: 'https://github.com/gchq/Palisade-common.git'
                     if (sh(script: "git checkout ${GIT_BRANCH_NAME}", returnStatus: true) == 0) {
-                        container('docker-cmds') {
-                            configFileProvider([configFile(fileId: "${env.CONFIG_FILE}", variable: 'MAVEN_SETTINGS')]) {
-                                sh 'mvn -s $MAVEN_SETTINGS install -P quick'
-                            }
-                        }
+                        COMMON_REVISION = "${GIT_BRANCH_NAME_LOWER}-BRANCH-SNAPSHOT"
```
Comment (pd104923) [inline Jenkinsfile:165]:
```suggestion
                        COMMON_REVISION = "BRANCH-${GIT_BRANCH_NAME_LOWER}-SNAPSHOT"
```

File: Jenkinsfile [hunk 2]
```diff
@@ -168,18 +185,24 @@ timestamps {
                     git branch: GIT_BRANCH_NAME, url: 'https://github.com/gchq/Palisade-examples.git'
                     container('docker-cmds') {
                         configFileProvider([configFile(fileId: "${env.CONFIG_FILE}", variable: 'MAVEN_SETTINGS')]) {
-                            sh 'mvn -s $MAVEN_SETTINGS install'
+                             if (IS_PR == "true") {
+                               sh "mvn -s ${MAVEN_SETTINGS} -D revision=${EXAMPLE_REVISION} -D common.revision=${COMMON_REVISION}  -D common.revision=${READERS_REVISION} -D common.revision=${CLIENTS_REVISION} deploy"
+                             } else {
+                               sh "mvn -s ${MAVEN_SETTINGS} -D revision=${EXAMPLE_REVISION} -D common.revision=${COMMON_REVISION}  -D common.revision=${READERS_REVISION} -D common.revision=${CLIENTS_REVISION} install"
+                             }
```
Comment (pd104923) [inline Jenkinsfile:192]:
```suggestion
                             if (IS_PR == "true") {
                               sh "mvn -s ${MAVEN_SETTINGS} -D revision=${EXAMPLE_REVISION} -D common.revision=${COMMON_REVISION}  -D readers.revision=${READERS_REVISION} -D clients.revision=${CLIENTS_REVISION} deploy"
                             } else {
                               sh "mvn -s ${MAVEN_SETTINGS} -D revision=${EXAMPLE_REVISION} -D common.revision=${COMMON_REVISION}  -D readers.revision=${READERS_REVISION} -D clients.revision=${CLIENTS_REVISION} install"
                             }
```

**Review by pd104923 — CHANGES_REQUESTED** [review]
Couple of minor changes in the Jenkinsfile

File: Jenkinsfile [hunk 3]
```diff
@@ -120,45 +120,71 @@ timestamps {
...
-                        container('docker-cmds') {
-                            configFileProvider([configFile(fileId: "${env.CONFIG_FILE}", variable: 'MAVEN_SETTINGS')]) {
-                                sh 'mvn -s $MAVEN_SETTINGS install -P quick'
-                            }
-                        }
+                        CLIENTS_REVISION = "BRANCH-${GIT_BRANCH_NAME_LOWER}-SNAPSHOT"
+                    }else{
+                      if (READERS_REVISION == "BRANCH-${GIT_BRANCH_NAME_LOWER}-SNAPSHOT"){
+                          sh "mvn -s ${MAVEN_SETTINGS} -D revision=${READERS_REVISION} -D common.revision=${READERS_REVISION} deploy"
+                          CLIENTS_REVISION = "BRANCH-${GIT_BRANCH_NAME_LOWER}-SNAPSHOT"
```
Comment (pd104923) [inline Jenkinsfile:186]:
```suggestion
                    } else {
                      if (READERS_REVISION == "BRANCH-${GIT_BRANCH_NAME_LOWER}-SNAPSHOT"){
                          sh "mvn -s ${MAVEN_SETTINGS} -D revision=${CLIENTS_REVISION} -D common.revision=${COMMON_REVISION} -D readers.revision=${READERS_REVISION} deploy"
                          CLIENTS_REVISION = "BRANCH-${GIT_BRANCH_NAME_LOWER}-SNAPSHOT"
```

**Review by pd104923 — CHANGES_REQUESTED** [review]
A small change to the clients prerequisite. The deploy for clients will need to know about the common revision as well as the readers revision.

**Review by pd104923 — APPROVED** [review]

**Review by developer01189998819991197253 — APPROVED** [review]

**Review by dev-958 — APPROVED** [review]

**Pull request merged**